[All Hosts] (DevX) updating tool articles for WXP with unified manifest (#4792)
OfficeDev/office-js-docs-pr: docs/testing/sideload-add-in-with-unified-manifest.md

---
title: Sideload Office Add-ins that use the unified manifest for Microsoft 365
description: Test your Office Add-in on Windows by sideloading.
ms.date: 09/19/2024
ms.localizationpriority: medium
---

# Sideload Office Add-ins that use the unified manifest for Microsoft 365

The process of sideloading an add-in that uses the [Unified manifest for Microsoft 365](../develop/json-manifest-overview.md) varies depending on the tool you want to use and on how the add-in project was created. 

> [!NOTE]

> > An add-in that uses the unified manifest can be sideloaded on Office on Windows, Version 2304 (Build 16320.20000) or later. Currently, it cannot be sideloaded on the web, Mac, or iPad.

Reviewer: ```suggestion
> An add-in that uses the unified manifest can be sideloaded on Office on Windows, Version 2304 (Build 16320.20000) or later. Currently, it can't be sideloaded on the web, Mac, or iPad.
```

> ## Sideload add-ins created with the Office Yeoman Generator (Yo Office)

Reviewer: ```suggestion
## Sideload add-ins created with the Yeoman generator for Office Add-ins (Yo Office)
```

Use the process described in [Sideload with a system prompt, bash shell, or terminal](#sideload-with-a-system-prompt-bash-shell-or-terminal).

## Sideload with the Teams Toolkit

1. First, *make sure Office desktop application that you want to sideload into is closed.*
1. In Visual Studio Code, open Teams Toolkit.
1. Required for Outlook only: in the **ACCOUNTS** section, verify that you're signed into Microsoft 365.

> 1. Select **View** | **Run** in Visual Studio Code. In the **RUN AND DEBUG** drop down menu, select one of these options as appropriate for your add-in.

Reviewer: ```suggestion
1. Select **View** | **Run** in Visual Studio Code. In the **RUN AND DEBUG** dropdown menu, select one of these options as appropriate for your add-in.
```

 
    - **Excel Desktop (Edge Chromium)**
    - **Outlook Desktop (Edge Chromium)**
    - **PowerPoint Desktop (Edge Chromium)**
    - **Word Desktop (Edge Chromium)**

1. Press F5. The project builds and a Node dev-server window opens. This process may take a couple of minutes and then the desktop version of the Office application that you selected opens. You can now work with your add-in. For an Outlook add-in, be sure you're working in the **Inbox** of *your Microsoft 365 account identity*.
1. To stop debugging and uninstall the add-in, select **Run** | **Stop Debugging** in Visual Studio Code. Closing the server window doesn't reliably stop the server and closing the Office application doesn't reliably cause Office to unacquire the add-in.

## Sideload with a system prompt, bash shell, or terminal

> 1. First, *make sure Office desktop application that you want to sideload into is closed.*

Reviewer: ```suggestion
1. First, *make sure the Office desktop application that you want to sideload into is closed.*
```

    ```command&nbsp;line
    npx office-addin-debugging stop <relative-path-to-unified-manifest>
    ``` 

### Sideload with the Teams Toolkit CLI (command-line interface) 

1. Manually create a zip package using the following steps.

    1. Open the unified manifest and scroll to the "icons" property. Note the relative path of the two image files.
    1. Use any zip utility to create a zip file that contains the unified manifest and the two image files. *The image files must have the same relative path in the zip file as they do in the project.* For example, if the relative path is "assets/icon-64.png" and "assets/icon-128.png", then you must include the "assets" folder with the two files in the zip package. 
    1. If the folder contains other files, such as image files used in the Office ribbon, remove these from the zip package. It should have only the two image files specified in the "icons" property (in addition to the manifest in the root of the zip package). 

1. In the root of the project, open a command prompt or bash shell and run the following commands.

    ```command&nbsp;line
    npm install -g @microsoft/teamsfx-cli

> 

Reviewer: I think separating the commands will make it easier for users to copy/paste/run

```suggestion
    ```
    ```command&nbsp;line
```
